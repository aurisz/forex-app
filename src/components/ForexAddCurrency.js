import React from 'react';
import PropTypes from 'prop-types';
import { Input, InputGroup, InputGroupAddon, Button } from 'reactstrap';

const ForexAddCurrency = ({
  options,
  value,
  onInputChange,
  onSubmitAddCurrency,
  onClickAddMoreCurrency,
  isAddMoreCurrency
}) => (
  <div className="forex-add-currency">
    {isAddMoreCurrency ? (
      <InputGroup>
        <Input
          type="select"
          name="selectedNewCurrency"
          value={value}
          onChange={onInputChange}
        >
          <option value="">Select Currency!</option>
          {options.map(curr => (
            <option key={curr} value={curr}>
              {curr}
            </option>
          ))}
        </Input>
        <InputGroupAddon addonType="append">
          <Button
            color="primary"
            onClick={onSubmitAddCurrency}
            disabled={value === ''}
          >
            Submit
          </Button>
        </InputGroupAddon>
      </InputGroup>
    ) : (
      <Button color="info" block onClick={onClickAddMoreCurrency}>
        (+) Add More Currencies
      </Button>
    )}
  </div>
);

ForexAddCurrency.propTypes = {
  options: PropTypes.arrayOf(PropTypes.string).isRequired,
  value: PropTypes.string.isRequired,
  onInputChange: PropTypes.func.isRequired,
  onSubmitAddCurrency: PropTypes.func.isRequired,
  onClickAddMoreCurrency: PropTypes.func.isRequired,
  isAddMoreCurrency: PropTypes.bool
};

ForexAddCurrency.defaultProps = {
  isAddMoreCurrency: false
};

export default ForexAddCurrency;
