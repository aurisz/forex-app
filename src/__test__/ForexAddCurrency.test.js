import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ForexAddCurrency from '../components/ForexAddCurrency';

describe('Forex Add Currency', () => {
  const testProps = {
    options: ['IDR'],
    value: 'EUR',
    onInputChange: jest.fn(),
    onSubmitAddCurrency: jest.fn(),
    onClickAddMoreCurrency: jest.fn(),
    isAddMoreCurrency: false
  };

  it('renders without crashing', () => {
    const wrapper = shallow(<ForexAddCurrency {...testProps} />);
    expect(wrapper.exists()).toBe(true);
  });
  it('should render button to add new currency', () => {
    const tree = renderer.create(<ForexAddCurrency {...testProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should render dropdown to select currency', () => {
    const tree = renderer
      .create(<ForexAddCurrency {...testProps} isAddMoreCurrency />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
