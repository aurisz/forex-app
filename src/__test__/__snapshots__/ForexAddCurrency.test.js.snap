// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Forex Add Currency should render button to add new currency 1`] = `
<div
  className="forex-add-currency"
>
  <button
    aria-label={null}
    className="btn btn-info btn-block"
    onClick={[Function]}
    type="button"
  >
    (+) Add More Currencies
  </button>
</div>
`;

exports[`Forex Add Currency should render dropdown to select currency 1`] = `
<div
  className="forex-add-currency"
>
  <div
    className="input-group"
  >
    <select
      className="form-control"
      name="selectedNewCurrency"
      onChange={[MockFunction]}
      value="EUR"
    >
      <option
        value=""
      >
        Select Currency!
      </option>
      <option
        value="IDR"
      >
        IDR
      </option>
    </select>
    <div
      className="input-group-append"
    >
      <button
        aria-label={null}
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Submit
      </button>
    </div>
  </div>
</div>
`;
